util/metrics/prometheus/prometheus.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright 2021 The TestGrid Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package prometheus
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/testgrid/util/metrics"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	dto "github.com/prometheus/client_model/go"
+)
+
+// Valuer extends a metric to include a report on its values.
+type Valuer interface {
+	metrics.Metric
+	Values() map[string]float64
+}
+
+type int64Metric struct {
+	name   string
+	fields map[string]bool
+	met    *prometheus.GaugeVec
+}
+
+// init sets up the Prometheus endpoint at this port and URL when the package is imported
```

Comment:
Should this 1) be an explicit initialization in the controller's main.go instead, and 2) be passed a port as an arg?

Reply:
I think that hinges on "Do you think there should be a new `--metrics-port` flag on all controllers?" I lean towards "don't bother (yet)", but not very strongly.

If the port can be changed between binaries, we'd want a way to change it via a flag that should be in each main.go program. Otherwise, we should enforce that it's on the same port for everyone, and I think this is the best pattern for that.

Reply:
I think "don't bother yet" seems fair, easy to add later!